### DISCUSSION TITLE
BL-1: Name Reactions

### DISCUSSION THREAD

**Discussion by LierB** [Global-Chem/global-chem#84] [General]
I have started with adding smiles to named_reactions_in_organic_synthesis.
For the Schotten-Baumann reaction and the Curtius rearrangement (up to now) there are more products possible dependent on the reactants. Schotten-Baumann will yield amides from amines or esters from alcohols. For now I have decided for one, but I think they would both be relevant I think (for Curtius even three products). 
What would you say, should we add more products with different reactants like 'schotten_baumann_reaction_amines' and 'schotten_baumann_reaction_alcohols'? Or would someone has a different idea how to deal with such cases?
Grateful for replies including suggestions and any other issues and thoughts regarding the named reactions.

Betty

**Comment by Sulstice** [comment]
Hi @LierB 

1884: Schotten found that he could prepare N-benzoyl piperidine from piperidine with NaOH. 
1886: Baumann said you could extend this to benzoic acid esters from alcohols.

My reference is Barbara Czako's et al. book "Named Reactions in Organic Synthesis" 

So yes, there are two main components to these reactions that have different functional groups and would be worth to separate. This list is proving to be difficult to complete so I propose we maybe switch out strategy. 

`smiles`, `smarts`, `smirks` can be combined to create atom classes where we can define R groups of relevance denoted as `[:1]`. This escapes using the [R] nomenclature in the SMILES but allows to retain reusable atom classes that we can reference back to which will help complete the list. 

For example to understand what I mean:

```python

atom_classes = {
   '1': 'hydrogen',
   '2': 'alkyl',
   '3': 'amide',
   '4: 'alcohol',
   '5': 'aryl',
   '6': 'benzyl',
   '7': 'amine',
}

```

We can perhaps use it like so in the SMILES string like so:

```
CC(C)([:47])C.CC(C)([:47]C(C)=O)C

```
The 4 and 7 meaning it can be alcohol or amine. What are your thoughts? Makes it easier to write.

**Reply by LierB** [reply]
Good morning @Sulstice 
just a short question to the `atom_classes`:
if we have a long list and we want to use e.g. `'12'` or `'3'` like this:

`CC(C)([:123])C.CC(C)([:123]C(C)=O)C`

how can we be sure it takes `'12'` or `'3'` and not `'1'` or `'23'`?
Maybe we need another notation to separate the options. Or did I misunderstand something here?
Sorry for just recognizing and mention it now.

**Comment by LierB** [comment]
Hey @Sulstice 

first thanks for history class ;) 

I agree with you that switching strategy could be beneficial for this list. Including R groups is a good idea and with your example it gets very clear what you mean.

We would need a whole list of all functional groups, which I could prepare (taking care of the order is not necessary, right).
I don't know if it would gets easier to write (actually I don't think so), but it would gets more general, opens new possibilities and allows for exactly these cases, where there are more possibilities of reaction products. 

I have assumed now, that we also include R groups in reactions where different products are not of relevance (just to shorten down the string?). Or do you think this would be an overkill and we should stick to the way we do it now for "normal" cases?

With this, what comes to my mind is, that for Schotten-Baumann the product depends on the reactant (2 reactions with two different reactants), but e.g. for Cannizzaro there are two products (with similar yields) in one reaction, which was not considered yet and could also be included. 
But here another consideration would be, that we probably would need to specify the groups more specifically like `'amine'` in primary, secondary or tertiary, but I think this could be included in `smiles` notation, or?
Last question, do we need to "translate" the numbers of the `atom_classes` into `smiles` that it could be backtranslated later. But if it's the case, e.g. alkyl would loose general character again, if we would just translate it to methyl. 

Ok maybe I'm anyways thinking too far away from the actual idea, please let me know about the points I mentioned.

**Comment by Safari222** [comment]
hey @LierB 
in response to: But here another consideration would be, that we probably would need to specify the groups more specifically like 'amine' in primary, secondary or tertiary, but I think this could be included in smiles notation, or?

I think this would be beneficial, as some reactions take place using specific amines (secondary, tertiary, primary) to yield different products. This is also the case to primary carbon, secondary carbon, tertiary carbon

Also (for the reactions that yield 2 different products) I feel it would be beneficial to note the major product as well as the minor product, and 50:50 racemization if that is the case. hope this makes sense!

**Reply by LierB** [reply]
Hey @Safari222 

yeah this was my thought to include them specifically, because in some cases actually just one/not all of them is/are possible to react. The question is now when we want to keep the `smiles` notation how deep do we want to go into R group specification. We have to clarify this. 

Of course this makes sense! For now I just wrote the major product, before @Sulstice had the idea with the R groups. But I also agree that sometimes both products should be mentioned. Same for reactants in some cases, e.g. the Biginelli reaction where all 3 reactants are relevant. 
Also racemates are are a good point, that we didn't consider yet, thanks!

**Comment by Sulstice** [comment]
@Safari222 Aziza, yeah it is. Go back to basics Sn2, Sn1, E1, E2. 

```
atom_classes = {
   '1': 'amine',
   '2': 'primary amine',
   '3': 'secondary amine',
   '4': 'tertiary amine',
   etc ...
}
```

Lets use the atom class idea in conjunction with an actual indexing that makes the most sense and serves the purpose of filling out this list. 
Biginelli is a multicomponent reaction and we are going to have others too, I bet Ugi is on this list off the top of my head.

For multicomponents we need to keep track a little bit. In the class object we need to have the attribute:

```
class NamedReactionsInOrganicSynthesis(object):
     
    __MULTICOMPONENT_REACTIONS__ = []

```

So we can keep track of those. The reactants for something like Biginelli:

<img width="530" alt="Screen Shot 2022-04-28 at 7 41 45 AM" src="https://user-images.githubusercontent.com/11812946/165744513-05b786e2-3948-4877-a0df-8c4df4cfb0f6.png">

We can do

```
R1.R2.R3.P1
```

In the smiles. This is saying that these 3 reactions to the product. For multicomponent reactions is there always only one dominant product is that a rule?

**Reply by LierB** [reply]
replying to @Sulstice 's question: "For multicomponent reactions is there always only one dominant product is that a rule?"

In the literature it's emphasized that more than 2 products (usually 3 to 7) are converted to 1 product (or a single product, or the final product) or that there are almost no side products. It's a one-pot reaction and almost all atoms are incorporated in the product. They are seen as efficient reactions belonging to green chemistry. 

here is the DOI https://doi.org/10.1016/B978-0-12-817825-6.00002-1 to following book chapter intro
<img width="570" alt="Bildschirmfoto 2022-04-29 um 10 15 29" src="https://user-images.githubusercontent.com/104396477/165910535-d74cb852-71f8-407b-a1bb-fac4e0b8f3d0.png">

here is a review introduction
<img width="385" alt="Bildschirmfoto 2022-04-29 um 10 26 04" src="https://user-images.githubusercontent.com/104396477/165910690-f82865ed-b882-4dcb-b7ae-5aab13edfd7d.png">
https://doi.org/10.1021/cr100233r 
I would have the pdf, if you want to read further.

I read more and they all agreed on that, so I would say it's a rule.

**Comment by Sulstice** [comment]
Yeah Doemling comes from Sharpless and Ugi. He's reported up to 10-component reactions once as I think a theory before in this paper 3-10 is a good number:

<img width="634" alt="Screen Shot 2022-04-29 at 7 29 12 AM" src="https://user-images.githubusercontent.com/11812946/165936896-114f8448-70bb-4ff8-9084-16337c7d4e16.png">

https://www.sciencedirect.com/science/article/abs/pii/S1367593100000958?via%3Dihub

Here's the figure. The rule for an MCR would be to be considered is:

1.) There is one dominant product meaning in the code the last molecule is going to the product regardless of how many components. 

```
chemical_list = {
   'ugi: 'R1.R2.R3.R4.P1"
   'biginelli: 'R1.R2.R3.P1"
}

```

2.) For multi functional groups that have similar mechanisms based on the reaction. 

```
CC(C)([:1]~[:2])C.CC(C)([:1]~[:2]~[:3])C(C)=O)C
```

This might or not work but we can have atom classes virtually bonded `~` borrowing from the SMIRKS or SMARTS language (I forget) that should be able to process into rdkit, use the virtual bond as a way to install information.

Another alternative is to start using CXSMILES which is used for CGenFF atom typing to store metadata about the atom on the SMILES string. An example would be for perfluorobutanoic acid:

```
O=C(O)C(F)(F)C(F)(F)C(F)(F)F |atomProp:0.atom_type.OG2D1:0.atom_idx.7:1.atom_type.CG2O2:1.atom_idx.5:2.atom_type.OG311:2.atom_idx.6:3.atom_type.CG312:3.atom_idx.3:4.atom_type.FGA2:4.atom_idx.4:5.atom_type.FGA2:5.atom_idx.8:6.atom_type.CG312:6.atom_idx.1:7.atom_type.FGA2:7.atom_idx.0:8.atom_type.FGA2:8.atom_idx.2:9.atom_type.CG302:9.atom_idx.9:10.atom_type.FGA3:10.atom_idx.10:11.atom_type.FGA3:11.atom_idx.11:12.atom_type.FGA3:12.atom_idx.12|
```

Your guys thoughts?

**Reply by LierB** [reply]
ok so even up to 10 reactants, wow!

1.) `chemical_list` totally makes sense to me for MCR, if we define that we always have just one product, yes.
But what I've mentioned earlier "e.g. for Cannizzaro there are two products (with similar yields) in one reaction, which was not considered yet and could also be included". If we still want to include such cases we would need a further attribute next to 
```
__MULTICOMPONENT_REACTIONS__ = []
```
or am I wrong? Because there it would be like
```
chemical_list = {
   'cannizzaro': 'R1.P1.P2'
}
```


2.) with `atom_classes` virtually bonded using this `~` sign might make sense to me, I have to think a bit
also CXSMILES which you used for CGenFF atom typing, I would need to have a closer look, I am not used to this notation at all

I think it gets more and more complex, but eventually it will include everything we need.
I mean organic chemistry is also complex;)

**Comment by Sulstice** [comment]
Yeah, so @LierB  let's be simple for now to get majority of the list done. We will have to truncate reactions that have more than one product while I decide algorithmically how to unravel that information but still retain a 1-D format. 

Any reactions that are

```python

chemical_list = {
   'reaction_1': 'R1.R2.P1',
   'reaction_2': 'R1.P1',
  'reaction_3': 'R1.R2.R3.R4.P1',
}

```

Any other reactions that have multiple dominant products I can handle them. CXSMILES is useful for storing meta data on an atom type but the string itself is ugly and pollutes the output. `~` might work, granted we are not using it for it's intended purpose but I think we can successfully parse the string into the RDKit compiler or we can do some funky string mining before it gets to that point. 

Have to see. Try completing as much as you can with only 1 product and let's see how many we get.

**Reply by LierB** [reply]
Ok right @Sulstice , I will try to get the list done as soon as I can;) I will stick to one product for now (like we started, just with adding more reactants if relevant). In cases where there are actually more products relevant, I will decide for one as before and will additionally include the second as a comment. Later, if we want to switch to another approach it will be easier to change, if we have the info there I think. Do you agree?

**Comment by Sulstice** [comment]
Yeah, I think that's the best course of action.

**Comment by Sulstice** [comment]
@LierB Thank you for production of the list. Give me a little second and let's do a mass analysis on the overall picture together. We will be using a software called MolPDF. I think we can use that to measure performance as a visual inspection for list generation. 

The PR is maintained here: https://github.com/Sulstice/global-chem/pull/108

**Comment by Sulstice** [comment]
@LierB 

So this is interesting and worth noting. Pericyclic reactions were a thing of the past for relic drug hunters but they still stand as valuable ring systems because of their spherical shape that is deep enough to fit into the pocket of a protein. 

For example: 

Adamantane:

![image](https://user-images.githubusercontent.com/11812946/171332641-40112d7a-6c59-4cbe-aa42-1667fb27dc72.png)

When I saw the Wurtz Coupling it reminds me that we should look for Valuable Pericylic Reactions.  Because this is just a cool application of that method. 

<img width="428" alt="Screen Shot 2022-06-01 at 1 15 36 AM" src="https://user-images.githubusercontent.com/11812946/171332738-97b71521-256a-4367-8152-abdbf598e683.png">

We would need someone in the field that knows these reactions to make a comment. Think I know someone.

**Reply by Sulstice** [reply]
Okay, let's have some fun. It is interesting to me that to drug designers who consider themselves good can distinguish between valuable molecules. Adamantane is on my list as a common fragment and in my arsenal, it also exists in others and is demonstrated by CGenFF. CGenFF can show the adamantane as something valuable for drug design would be accounted for. So when I test it, it is fine. But pericylic reactions especially wurtz coupling has died out of value. Especially something as strained cyclobutane. 

The effective parameter worth looking at:

```
CG3C31 CG3RC1 CG3C31   77.35    111.00    8.00   2.56100 ! ***** , from CG3C31 CG3C31 CG3C31, penalty= 48
```

Could be relatively and easy target. 

Name: `bicyclo[1.1.0]butane`
SMILES: `C12CC1C2`

CGenFF:
```
* Toppar stream file generated by
* CHARMM General Force Field (CGenFF) program version 2.5
* For use with CGenFF version 4.5
*

read rtf card append
* Topologies generated by
* CHARMM General Force Field (CGenFF) program version 2.5
*
36 1

! "penalty" is the highest penalty score of the associated parameters.
! Penalties lower than 10 indicate the analogy is fair; penalties between 10
! and 50 mean some basic validation is recommended; penalties higher than
! 50 indicate poor analogy and mandate extensive validation/optimization.

RESI *****          0.000 ! param penalty=  48.000 ; charge penalty=  20.782
GROUP            ! CHARGE   CH_PENALTY
ATOM C1     CG3RC1 -0.084 !   20.612
ATOM C2     CG3C31 -0.186 !   20.782
ATOM C3     CG3RC1 -0.084 !   20.612
ATOM C4     CG3C31 -0.186 !   20.782
ATOM H1     HGA1    0.090 !    0.000
ATOM H2     HGA2    0.090 !    1.874
ATOM H3     HGA2    0.090 !    1.874
ATOM H4     HGA1    0.090 !    0.000
ATOM H5     HGA2    0.090 !    1.874
ATOM H6     HGA2    0.090 !    1.874

BOND C1   C2  
BOND C2   C3  
BOND C1   C3  
BOND C3   C4  
BOND C1   C4  
BOND C1   H1  
BOND C2   H2  
BOND C2   H3  
BOND C3   H4  
BOND C4   H5  
BOND C4   H6  

END

read param card flex append
* Parameters generated by analogy by
* CHARMM General Force Field (CGenFF) program version 2.5
*

! Penalties lower than 10 indicate the analogy is fair; penalties between 10
! and 50 mean some basic validation is recommended; penalties higher than
! 50 indicate poor analogy and mandate extensive validation/optimization.

BONDS

ANGLES
CG3C31 CG3RC1 CG3C31   77.35    111.00    8.00   2.56100 ! ***** , from CG3C31 CG3C31 CG3C31, penalty= 48

DIHEDRALS
CG3RC1 CG3C31 CG3RC1 CG3C31     0.1500  3     0.00 ! ***** , from CG3RC1 CG3C31 CG3RC1 CG3C52, penalty= 26.5
HGA2   CG3C31 CG3RC1 CG3C31     0.1950  3     0.00 ! ***** , from HGA2 CG3C31 CG3RC1 CG3C52, penalty= 26.5
CG3C31 CG3RC1 CG3RC1 CG3C31     0.1500  3     0.00 ! ***** , from CG3C31 CG3RC1 CG3RC1 CG3C52, penalty= 26.5

IMPROPERS

END
RETURN
```

Name: Adamantane 
SMILES: C12CC3CC(C2)CC(C3)C1

CGenFF:
```
* Toppar stream file generated by
* CHARMM General Force Field (CGenFF) program version 2.5
* For use with CGenFF version 4.5
*

read rtf card append
* Topologies generated by
* CHARMM General Force Field (CGenFF) program version 2.5
*
36 1

! "penalty" is the highest penalty score of the associated parameters.
! Penalties lower than 10 indicate the analogy is fair; penalties between 10
! and 50 mean some basic validation is recommended; penalties higher than
! 50 indicate poor analogy and mandate extensive validation/optimization.

RESI *****          0.000 ! param penalty=   0.000 ; charge penalty=   0.000
GROUP            ! CHARGE   CH_PENALTY
ATOM C1     CG311  -0.093 !    0.000
ATOM C2     CG321  -0.178 !    0.000
ATOM C3     CG311  -0.093 !    0.000
ATOM C4     CG321  -0.178 !    0.000
ATOM C5     CG311  -0.093 !    0.000
ATOM C6     CG321  -0.178 !    0.000
ATOM C7     CG321  -0.178 !    0.000
ATOM C8     CG311  -0.093 !    0.000
ATOM C9     CG321  -0.178 !    0.000
ATOM C10    CG321  -0.178 !    0.000
ATOM H1     HGA1    0.090 !    0.000
ATOM H2     HGA2    0.090 !    0.000
ATOM H3     HGA2    0.090 !    0.000
ATOM H4     HGA1    0.090 !    0.000
ATOM H5     HGA2    0.090 !    0.000
ATOM H6     HGA2    0.090 !    0.000
ATOM H7     HGA1    0.090 !    0.000
ATOM H8     HGA2    0.090 !    0.000
ATOM H9     HGA2    0.090 !    0.000
ATOM H10    HGA2    0.090 !    0.000
ATOM H11    HGA2    0.090 !    0.000
ATOM H12    HGA1    0.090 !    0.000
ATOM H13    HGA2    0.090 !    0.000
ATOM H14    HGA2    0.090 !    0.000
ATOM H15    HGA2    0.090 !    0.000
ATOM H16    HGA2    0.090 !    0.000

BOND C1   C2  
BOND C2   C3  
BOND C3   C4  
BOND C4   C5  
BOND C5   C6  
BOND C1   C6  
BOND C5   C7  
BOND C7   C8  
BOND C8   C9  
BOND C3   C9  
BOND C8   C10 
BOND C1   C10 
BOND C1   H1  
BOND C2   H2  
BOND C2   H3  
BOND C3   H4  
BOND C4   H5  
BOND C4   H6  
BOND C5   H7  
BOND C6   H8  
BOND C6   H9  
BOND C7   H10 
BOND C7   H11 
BOND C8   H12 
BOND C9   H13 
BOND C9   H14 
BOND C10  H15 
BOND C10  H16 

END

read param card flex append
* Parameters generated by analogy by
* CHARMM General Force Field (CGenFF) program version 2.5
*

! Penalties lower than 10 indicate the analogy is fair; penalties between 10
! and 50 mean some basic validation is recommended; penalties higher than
! 50 indicate poor analogy and mandate extensive validation/optimization.

BONDS

ANGLES

DIHEDRALS

IMPROPERS

END
RETURN
```
